src/Microsoft.IdentityModel.Tokens/Base64UrlEncoder.cs

```diff
@@ -249,8 +269,76 @@ private unsafe static byte[] UnsafeDecode(string str)
                 }
             }
         }
+
+        internal static unsafe byte[] UnsafeDecode(char[] str)
+        {
+            int mod = str.Length % 4;
+            if (mod == 1)
+                throw LogHelper.LogExceptionMessage(new FormatException(LogHelper.FormatInvariant(LogMessages.IDX10400, str)));
+
+            bool needReplace = false;
+            int decodedLength = str.Length + (4 - mod) % 4;
```

Comment:
nit: Consider adding summary for what this method is doing and comments for some of the logic.